turbopack/crates/turbopack-ecmascript/src/references/amd.rs

```diff
@@ -123,12 +125,11 @@ pub enum AmdDefineFactoryType {
     Value,
 }
 
-#[turbo_tasks::value(shared)]
-#[derive(Debug)]
+#[derive(PartialEq, Eq, Serialize, Deserialize, TraceRawVcs, ValueDebugFormat, NonLocalValue)]
 pub struct AmdDefineWithDependenciesCodeGen {
     dependencies_requests: Vec<AmdDefineDependencyElement>,
     origin: ResolvedVc<Box<dyn ResolveOrigin>>,
```

Comment:
I guess all CodeGens have the same `origin`. Maybe we could just pass it to the `code_generation` method instead of storing it many times

Reply:
👍 
That is btw exactly what I did in my run-analysis-only-once-for-SSR-and-Browser
Because these two modules would have the exact same references and codegens, except for that origin.

Reply:
However, this is only the case for this one specific `CodeGen`.
All other code gen snippets that need the `origin` are actually `References`, and are untouched by this PR (and they need to store the origin anyway)